### PR TITLE
Return instance URI for Form EDIT action

### LINKED ISSUE #5008: FormEntryActivity does not return instance uri after finishing a form
#### Software and hardware versions 
Collect v2021.3.4

#### Problem description
`FormEntryActivity` does not return instance uri after finishing a form. The problem seems to be caused by the fact that we hide `FormEntryActivity` behind `FormUriActivity` after merging https://github.com/getodk/collect/commit/0e927d7a8bdc3254690cf27b49083235bf95214e and `FormUriActivity` does not return the value returned by `FormEntryActivity` it is just ignored.

#### Other information 
The problem has been originally reported on the forum: https://forum.getodk.org/t/data-uri-is-null-in-onactivityresult/36548

### PR DESCRIPTION
**Pull request by seadowg**
Closes #5008

#### What has been done to verify that this works as intended?

Updated test.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Should just fix the problem. The VIEW/EDIT action for forms is the only thing that's had any changes.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -288,7 +288,6 @@ the specific language governing permissions and limitations under the License.
         Do not use the activity directly as its name/package might change. -->
         <activity
             android:name=".external.FormUriActivity"
-            android:noHistory="true"
```
Comment (seadowg) [inline collect_app/src/main/AndroidManifest.xml:291]: As far as I can see we aren't using this any longer now we explicitly call `finish()` in all cases. `noHistory` doesn't work with activity results as the calling Activity is finished as soon as the user hits the next Activity.

File: collect_app/src/main/java/org/odk/collect/android/external/FormUriActivity.kt [hunk 2]
```diff
@@ -14,14 +15,20 @@ import org.odk.collect.android.utilities.ApplicationConstants
...
+class FormUriActivity : ComponentActivity() {
 
     @Inject
     lateinit var currentProjectProvider: CurrentProjectProvider
 
     @Inject
     lateinit var projectsRepository: ProjectsRepository
 
+    private val openForm =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/external/FormUriActivity.kt:27]: Seemed like a good opportunity to move to the new API for this.

**Review by grzesiek2010 — APPROVED** [review]

**Comment by grzesiek2010** [comment]
I've added `needs testing` label but I don't know if it's something our QA team is able to verify, probably not. @kkrawczyk123 @srujner please at least take a look to make sure there is no regression when starting forms especially using external apps like our tester: https://github.com/grzesiek2010/collectTester

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids 5.1 and 8.1

Verified cases:
- download, fill and send forms form/to Central, Drive, Ona and Kobo 
- use CollectTester, download, fill form, send
- use CollectCounter
- download the same form twice

**Comment by srujner** [comment]
**Tested with success!**
Verified on Androids 10 and 11:

- use CollectTester, download, fill form, send
- use CollectCounter
- download the same form twice
- download, fill and send forms

**Pull request merged**